gigascience/gigadb-website
# Documentation about EM queries and reports

# Pull request for issue: #1047 Learn EM and Improve EM query (time-box)

This is a pull request for the following functionalities:

* Queries created in Editorial Manager for outputting information required by GigaReview.
* Documentation describing the queries and reports created on Editorial Manager for the Peer Review System

## How to test?

1. Log into the [GigaScience Editorial Manager website](http://giga.edmgr.com/) using @pli888's EM credentials which are available from the Gitlab cngb-infra project.
2. On the Enterprise Analytics Reporting web page, you will see the queries for the GigaReview system in the `Peer_Review_System` directory. Hover over the `Manuscripts_Previous_Week_PL` link and select the `pen` icon.
3. Using the `Data Sources`, `Fields` and the `Filter` tabs, you can see how this query has been created. Data Sources define the tables or views which are queried. Fields are those columns whose data are outputted in the reports. Filter provides information as to how the data in EM have been restricted to extract a range of data to be outputted in a report.
4. The `Preview` tab can be selected to see the contents of a report. 
5. Log into the Tech Team's SFTP server using the credentials available from the GitLab `cngb-infra` project. Change directory to the `editorialmanager` directory to see the reports that been uploaded by EM for each query.

## Any issues with implementation?

Even with `weekly` queries, there is a chance that there are no results in reports if no papers have been accepted for publication that week. In this case, change query to retrieve data from past month.

## Any changes to documentation?

The `docs/prs/REPORTS.md` has been updated with documentation about the EM queries and reports.
